### DISCUSSION TITLE
can't seem to find a way to make formatted date sortable

### DISCUSSION THREAD

**Discussion by snapey** [Power-Components/livewire-powergrid#316] [Q&A] [answered]
I can't seem to find out how to create a column containing created_at into a sortable column.

If I want the date with a custom format then I have to create a new column with a new name;
```
            ->addColumn('created_at_formatted', function(Organisation $organisation) { 
                return Carbon::parse($organisation->created_at)->format('d/m/Y H:i');
            });
```
but then when adding sortable to this column, and clicking the column heading, of course I get;
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'created_at_formatted' in 'order clause' 
```

If I just leave the column as `created_at` then the sort works but it ignores the closure on the addColumn meaning that I cannot display the custom date format.

I was really hoping that I could specify the sort column like this;

```
            Column::add()
                ->title(__('CREATED AT'))
                ->field('created_at_formatted')
                ->sortable('created_at'),
```
specifying the column to be sorted in the sortable method.

or

have PG accept the formatting of the created_at column without having to specify a new name

**Comment by snapey** [comment, marked as answer]
After source diving I found the solution
```
            Column::add()
                ->title(__('CREATED AT'))
                ->field('created_at_formatted','created_at')
                ->sortable(),
```
by specifying the 'source' data in the second parameter of the field() function

This could be clearer in the docs

Also think its odd that its not possible to overwrite the format of the column without giving it a new name

**Reply by snapey** [reply]
Had to come here and rediscover my own answer!

**Reply by luanfreitasdev** [reply]
https://livewire-powergrid.com/#/table/add-columns
